src/factory/ChartFactory.js

```diff
@@ -38,7 +53,10 @@ function ChartFactory({
   comparisonData,
   referenceData,
   profiles,
-  ...visualProps
+  /**
+   * Custom properties to apply to the chart
+   */
+  ...chartProps
```

Comment:
1. Spreading `props` like this is dangerous, since each chart will have its own set props that may change at any time. Soon or later they're going to clash with the rest of variables
2. We're mixing some chart props in `definition` e.g. `horizontal` while the rest are declared here. We need to have  a single place for these.

```js
function ChartFactory({
  theme,
  definition: {
    id,
    type: visualType,
    label,
    reference: { label: propReferenceLabel } = {},
    aggregate,
    unique,
    subtitle,
    description,
    locale = 'en-GB',
    customUnit = '',
    /**
    /**
     * The rest of the props are going to be considered as:
     *  Intl.NumberFormatOptions
     *
     * Omit unit simce its an experimental NumberFormat option.
     */
    unit,
    ...numberFormat
  },
  data,
  isComparison,
  comparisonData,
  referenceData,
  profiles,
  definitionProps 
}) {

....

const { horizontal, width, height, offset, ...visualProps } =  definitionProps || {}
```

Reply:
I would move horizontal back to definition. 
Unless we want to change that in the hurumap-data plugin.

```
const { width, height, offset, ...visualProps } =  definitionProps || {}
```

@kilemensi

Reply:
I know @karimkawambwa ... That's why I was playing the waiting game on introducing these changes. But we need to sort this out once and for all so I'm for changing it in HURUmap plugin as well